client/src/main.rs

```diff
@@ -284,7 +284,12 @@ fn run() -> Result<(), Error> {
                 .into_ip_address().unwrap();
             let port = metrics_settings.port.unwrap_or(s::DEFAULT_METRICS_PORT);
             info!("Starting metrics server listening on port {}", port);
-            other_futures.push(metrics_server(Arc::clone(&consensus), bind, port, metrics_settings.password)?);
+            if settings.network.protocol == s::Protocol::Wss {
+                let tls_settings = settings.network.tls.expect("This should always exist for protocol Wss");
```

Comment:
I would recommend a message like `"Using Wss protocol, but TLS settings missing`". Otherwise the error message one would get isn't really helpful. And you can definitely set protocol to wss, but not have a `[tls]` section in the config file.

Reply:
I was using that message mostly for internal purposes, given that at that point, it shouldn't be possible to be using `wss` and not have a `[tls]` section because of [this previous check](https://github.com/nimiq/core-rs/blob/master/client/src/main.rs#L211-L218).

But I still think it's better to have a descriptive message that an user can understand, even if it theoretically should never happen, [lest we end up with a brother to the Root](https://multicians.org/hodie-natus-est.html) :smile: :

>"The code was in a "can't happen" case, which was why it was left in shipping code."